migration/precheck_shim.go

```diff
@@ -97,7 +115,15 @@ func (s *precheckShim) ControllerBackend() (PrecheckBackendCloser, error) {
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	return &precheckShim{st}, nil
+	rSt, err := st.Resources()
+	if err != nil {
+		st.Close()
```

Comment:
So we close state if we encounter an error here... Where do we close it if we succeed? I have not seen any close calls.. but may be I just don't have eagle eyes :)